### PR TITLE
fix:remove deprecations

### PR DESCRIPTION
**Pull request by vishal-cd**
## what
* Added `subscription_id` for provider in root module
* Also updated the modules release version in root modules
* Fixed automerge workflow. 

## why
* subscription_id is compulsory to add for terraform planning

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -5,31 +5,49 @@
...
 # Name of this project
-name : Terraform AZURE VPN
+name: Terraform AZURE FLEXIBLE MYSQL
 
 # License of this project
 license: "APACHE"
 
 # Canonical GitHub repo
-github_repo: clouddrove/terraform-azure-vpn
+github_repo: clouddrove/terraform-azure-flexible-mysql
```
Comment (archit-cd) [inline README.yaml]: Flexible mysql?

File: README.yaml [hunk 2]
```diff
@@ -5,31 +5,49 @@
 #
 
 # Name of this project
-name : Terraform AZURE VPN
+name: Terraform AZURE FLEXIBLE MYSQL
```
Comment (archit-cd) [inline README.yaml]: Flexible Mysql?

File: README.yaml [hunk 3]
```diff
@@ -5,31 +5,49 @@
...
+    url: https://github.com/clouddrove/terraform-azure-labels
+    description: Provides resource tagging.
 
 #  description of this project
 description: |-
-  Terraform module to create vpn resource on AZURE.
-# extra content
-include:
-  - "terraform.md"
+  Terraform module to create flexible-mysql resource on AZURE.
```
Comment (archit-cd) [inline README.yaml]: Flexible Mysql?

File: versions.tf [hunk 4]
```diff
@@ -9,5 +9,9 @@ terraform {
       source  = "hashicorp/azurerm"
       version = ">=3.0.0"
     }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.1"
```
Comment (archit-cd) [inline versions.tf]: can we make it `>=` ?

File: README.yaml (hunk 1 above)
Comment (vishal-cd) [reply README.yaml]: changed it to vpn

File: README.yaml (hunk 3 above)
Comment (vjdbj) [reply README.yaml]: changed it to vpn

File: README.yaml (hunk 2 above)
Comment (vishal-cd) [reply README.yaml]: changed it to VPN

File: README.yaml (hunk 3 above)
Comment (vishal-cd) [reply README.yaml]: change it to vpn

File: versions.tf (hunk 4 above)
Comment (vishal-cd) [reply versions.tf]: done

File: output.tf [hunk 5]
```diff
@@ -1,18 +1,19 @@
 output "vpn_gw_id" {
-  value       = join("", azurerm_virtual_network_gateway.vpngw.*.id)
+  value       = length(azurerm_virtual_network_gateway.vpngw) > 0 ? azurerm_virtual_network_gateway.vpngw[0].id : null
```
Comment (archit-cd) [inline output.tf]: Any reason we are using the length function? Also it is not used correctly in all output.

File: variables.tf [hunk 6]
```diff
@@ -4,13 +4,6 @@ variable "resource_group_name" {
   description = "A container that holds related resources for an Azure solution"
 }
 
-variable "location" {
```
Comment (archit-cd) [inline variables.tf:7]: Why are we removing it?

Comment (vishal-cd) [reply variables.tf:7]: Its is not being used.

File: output.tf (hunk 5 above)
Comment (vishal-cd) [reply output.tf]: Used length function so that It checks for the empty value but have changed it with more reliable try function.

**Pull request merged**